ext/reflection/php_reflection.c

```diff
@@ -2108,6 +2114,170 @@ ZEND_METHOD(reflection_function, getExtensionName)
+	zend_generator *root_generator;
+	zend_execute_data *ex_backup = EG(current_execute_data);
+	zend_execute_data *ex = generator->execute_data;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "|l", &options) == FAILURE) {
+		return;
+	}
+
+	REFLECTION_CHECK_VALID_GENERATOR(ex)
+
+	root_generator = zend_generator_get_current(generator);
+
+	if (generator == root_generator) {
+		generator->execute_data->prev_execute_data = NULL;
+	} else {
+		generator->execute_fake.prev_execute_data = NULL;
+		root_generator->execute_data->prev_execute_data = &generator->execute_fake;
+	}
+
+	EG(current_execute_data) = root_generator->execute_data;
+	zend_fetch_debug_backtrace(return_value, 0, options, 0);
+	EG(current_execute_data) = ex_backup;
+}
+/* }}} */
+
+/* {{{ proto public void ReflectionGenerator::getExecutingLine() */
+ZEND_METHOD(reflection_generator, getExecutingLine)
+{
+	zend_generator *generator = (zend_generator *) Z_OBJ(Z_REFLECTION_P(getThis())->obj);
+	zend_execute_data *ex = generator->execute_data;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	REFLECTION_CHECK_VALID_GENERATOR(ex)
+
+	ZVAL_LONG(return_value, ex->opline->lineno);
+}
+/* }}} */
+
+/* {{{ proto public void ReflectionGenerator::getExecutingFile() */
+ZEND_METHOD(reflection_generator, getExecutingFile)
+{
+	zend_generator *generator = (zend_generator *) Z_OBJ(Z_REFLECTION_P(getThis())->obj);
+	zend_execute_data *ex = generator->execute_data;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	REFLECTION_CHECK_VALID_GENERATOR(ex)
+
+	ZVAL_STR_COPY(return_value, ex->func->op_array.filename);
+}
+/* }}} */
+
+/* {{{ proto public void ReflectionGenerator::getFunction() */
+ZEND_METHOD(reflection_generator, getFunction)
+{
+	zend_generator *generator = (zend_generator *) Z_OBJ(Z_REFLECTION_P(getThis())->obj);
+	zend_execute_data *ex = generator->execute_data;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	REFLECTION_CHECK_VALID_GENERATOR(ex)
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	if (zend_string_equals_literal(ex->func->op_array.function_name, "{closure}")) {
+		zend_create_closure(return_value, ex->func, ex->func->common.scope, ex->called_scope, Z_OBJ(ex->This) ? &ex->This : NULL);
+		return;
+	}
+
+	if (Z_OBJ(ex->This)) {
+		zval func;
+		ZVAL_STR_COPY(&func, ex->func->op_array.function_name);
+
+		array_init(return_value);
+		Z_ADDREF(ex->This);
+		zend_hash_index_add_new(Z_ARR_P(return_value), 0, &ex->This);
+		zend_hash_index_add_new(Z_ARR_P(return_value), 1, &func);
+	} else if (ex->func->op_array.scope) {
+		zval class;
+		zval func;
+		ZVAL_STR_COPY(&class, ex->func->op_array.scope->name);
+		ZVAL_STR_COPY(&func, ex->func->op_array.function_name);
+
+		array_init(return_value);
+		zend_hash_index_add_new(Z_ARR_P(return_value), 0, &class);
+		zend_hash_index_add_new(Z_ARR_P(return_value), 1, &func);
+	} else {
+		ZVAL_STR_COPY(return_value, ex->func->op_array.function_name);
+	}
```

Comment:
I feel like this should maybe return a `ReflectionFunctionAbstract` rather than a callback. Would be more in line with how other reflection methods work.